### PR TITLE
ROE 2415 Add params to individual MO urls

### PR DESCRIPTION
**Pull request by dlloyd1-ch**
### JIRA link
https://companieshouse.atlassian.net/browse/ROE-2415

### Change description
Add url params to urls (get, post, update, remove) for the Individual MO page

### Work checklist

- [x] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: test/controllers/beneficial.owner.individual.controller.spec.ts [hunk 1]
```diff
@@ -856,7 +856,7 @@ describe("BENEFICIAL OWNER INDIVIDUAL controller", () => {
   });
 
   describe("POST with url params tests", () => {
-    test(`redirects to ${BENEFICIAL_OWNER_TYPE_PAGE} page`, async () => {
+    test.only(`redirects to ${BENEFICIAL_OWNER_TYPE_PAGE} page`, async () => {
```
Comment (markpit) [inline test/controllers/beneficial.owner.individual.controller.spec.ts:859]: .only needs to be removed

File: test/controllers/beneficial.owner.type.controller.spec.ts [hunk 2]
```diff
@@ -181,7 +181,7 @@ describe("BENEFICIAL OWNER TYPE controller", () => {
       expect(resp.text).toContain(BENEFICIAL_OWNER_TYPE_PAGE_HEADING);
       expect(resp.text).not.toContain(PAGE_TITLE_ERROR);
       expect(resp.text).toContain(NEXT_PAGE_URL);
-      expect(mockGetUrlWithParamsToPath).toHaveBeenCalledTimes(3);
+      expect(mockGetUrlWithParamsToPath).toHaveBeenCalledTimes(4);
       expect(mockGetUrlWithParamsToPath.mock.calls[0][0]).toEqual(config.BENEFICIAL_OWNER_TYPE_WITH_PARAMS_URL);
       expect(mockGetUrlWithParamsToPath.mock.calls[1][0]).toEqual(config.BENEFICIAL_OWNER_TYPE_SUBMIT_WITH_PARAMS_URL);
       expect(mockGetUrlWithParamsToPath.mock.calls[2][0]).toEqual(config.BENEFICIAL_OWNER_INDIVIDUAL_WITH_PARAMS_URL);
```
Comment (mattch1) [inline test/controllers/beneficial.owner.type.controller.spec.ts:187]: I cannot see the line for managing officer here.

File: src/routes/index.ts [hunk 3]
```diff
@@ -250,6 +250,12 @@ router.post(config.MANAGING_OFFICER_URL, authentication, navigation.hasBeneficia
 router.post(config.MANAGING_OFFICER_URL + config.ID, authentication, navigation.hasBeneficialOwnersStatement, ...validator.managingOfficerIndividual, checkValidations, managingOfficerIndividual.update);
 router.get(config.MANAGING_OFFICER_URL + config.REMOVE + config.ID, authentication, navigation.hasBeneficialOwnersStatement, managingOfficerIndividual.remove);
 
+router.get(config.MANAGING_OFFICER_WITH_PARAMS_URL, authentication, navigation.hasBeneficialOwnersStatement, managingOfficerIndividual.get);
```
[the remark this answers is not in the document]
Comment (mattch1) [reply src/routes/index.ts:253]: Would it be easier to keep this as it is for when we come to remove the FF along with the original urls here?

[the remark this answers is not in the document]
Comment (markpit) [reply src/routes/index.ts:253]: yes I guess so, the overarching reason is to make it easy to find and remove them so as long as they are next to the original non params version should be ok. Can leave as is.

File: test/controllers/beneficial.owner.type.controller.spec.ts (hunk 2 above)
Comment (dlloyd1-ch) [reply test/controllers/beneficial.owner.type.controller.spec.ts:187]: Sorted

File: test/controllers/beneficial.owner.individual.controller.spec.ts (hunk 1 above)
Comment (dlloyd1-ch) [reply test/controllers/beneficial.owner.individual.controller.spec.ts:859]: This file was committed in error. Have reverted change.

**Review by markpit — APPROVED** [review]

File: src/controllers/managing.officer.controller.ts [hunk 4]
```diff
@@ -7,24 +7,35 @@ import {
...
-  updateManagingOfficer(req, res, next, BENEFICIAL_OWNER_TYPE_URL, true);
+  updateManagingOfficer(req, res, next, getNextPageURL(req), true);
 };
 
 export const remove = (req: Request, res: Response, next: NextFunction) => {
-  removeManagingOfficer(req, res, next, BENEFICIAL_OWNER_TYPE_URL, true);
+  removeManagingOfficer(req, res, next, getNextPageURL(req), true);
+};
+
+const getNextPageURL = (req: Request): string => {
```
Comment (mwestacott) [inline src/controllers/managing.officer.controller.ts:35]: How about changing the name to match what @markpit suggested and implemented in BO Other and I adopted in BO Gov.

**Review by mattch1 — APPROVED** [review]

**Pull request merged**